### ISSUE TITLE
ActiveStorage Blobs Sigd's changed after app update

### ISSUE THREAD

**Issue by matuszewskijan** [rails/rails#40266]
### Steps to reproduce
I am not sure how I can reproduce this issue.

### Expected behavior
Sigd's never changed?

### Actual behavior
Hi, I've pushed new version of my Rails application using `Capistrano` and suddenly after the deploy all of my `ActiveText` attachments disappeared. The update consisted of some `SMTP` configuration changes and also some changes Capistrano `linked_dirs` - maybe something here broken my current setup.

I've found out that all of my images are present in `/storage` folder and I can list all of them using `ActiveStorage::Blob.all`.
Strange thing is that saved by `ActionText` sgids differ from current `ActiveStorage` sgids:
```ruby
Page.first.content.body.attachments.first.full_attributes.slice('filename', 'sgid')
{"filename"=>"vakantiehuis_duitsland_wandelen.jpg", "sgid"=>"BAh7CEkiCGdpZAY6BkVUSSI0Z2lkOi8vdHJhdmVsL0FjdGl2ZVN0b3JhZ2U6OkJsb2IvMzIwP2V4cGlyZXNfaW4GOwBUSSIMcHVycG9zZQY7AFRJIg9hdHRhY2hhYmxlBjsAVEkiD2V4cGlyZXNfYXQGOwBUMA==--568ed527c296e4585248bf86bfd566d1e95171f8"}
ActiveStorage::Blob.find_by(filename: 'vakantiehuis_duitsland_wandelen.jpg').attachable_sgid
"BAh7CEkiCGdpZAY6BkVUSSI0Z2lkOi8vdHJhdmVsL0FjdGl2ZVN0b3JhZ2U6OkJsb2IvMzIwP2V4cGlyZXNfaW4GOwBUSSIMcHVycG9zZQY7AFRJIg9hdHRhY2hhYmxlBjsAVEkiD2V4cGlyZXNfYXQGOwBUMA==--ba99ca867ed6255be92dcae124afeb8e0b731707"
```
I am sure that both files are the same files. Do you know how it's possible that `sgids` got regenerated? Is it normal? If so when it happen? 
I wasn't able to find any information about this kind of problem anywhere else :sob: 

### System configuration
**Rails version**:
6.0.2
**Ruby version**:
2.6.5

**Comment by eileencodes** [comment]
What version did you upgrade from? Can you try making a new application in that version, add some blobs, and then upgrade to the version you're now on and see if that reproduces the issue outside your application?

**Comment by matuszewskijan** [comment]
Actually I've found the source of the issue. We've made some changes to our encrypted credentials in the update, by mistake also `secret_key_base` got changed. It seems that when `secret_key_base` is changing then all urls to Blobs are regenerated. 

I can't find exact place in codebase where it's done but for sure it was the source of the issue as I was able to reproduce it by changing and reverting `secret_key_base` many times.
